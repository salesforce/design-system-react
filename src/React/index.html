<!doctype html public "library">
<meta charset=utf-8/>
<head>
	<title>Selectlist</title>
	<link rel="stylesheet" href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.5/css/bootstrap.min.css">
</head>
<body>
<script src="/dist/react/build.js"></script>
</body>
</html>