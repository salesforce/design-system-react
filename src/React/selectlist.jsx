//test

var React = require('react');
import {SelectlistCore} from "../SelectlistCore";
var DropdownButton = require('react-bootstrap/lib/DropdownButton');
var MenuItem = require('react-bootstrap/lib/MenuItem');

var DATA = {
	selected: 'None selected',
	id: 'selectlist1',
	items: [
		{ id: 0, name: 'tacos', type: 'mexican' },
		{ id: 1, name: 'burrito', type: 'mexican' },
		{ id: 2, name: 'tostada', type: 'mexican' },
		{ id: 3, name: 'hush puppies', type: 'southern' }
	]
};

var menuItemSelected = (eventKey, href, target)=> {
	DATA.selected = DATA.items[eventKey].name;
};
//Selectlist.handleMenuItemClicked
var menuItems = DATA.items.map( (menuItem) => {
	return <MenuItem eventKey={menuItem.id}>{menuItem.name}</MenuItem>;
});

var Selectlist = React.createClass({
	getInitialState: function() {
		return {
			selectedKey: undefined
		};
	},

	handleMenuItemClicked: function(eventKey, href, target) {
		this.setState({
			selectedKey: eventKey
		});
	},

	menuItems: function() {
		return DATA.items.map( (menuItem) => {
			return <MenuItem eventKey={menuItem.id} onSelect={this.handleMenuItemClicked}>{menuItem.name}</MenuItem>;
		});
	},

	selected: function() {
		return typeof(this.state.selectedKey) === 'undefined' ? 'None Selected' : DATA.items[this.state.selectedKey].name;
	},

	render () {
		return (
			<div>
				<h1>SelectList</h1>
				<ul>{this.menuItems}</ul>
				<DropdownButton title={this.selected()} key={DATA.id}>{this.menuItems()}</DropdownButton>
			</div>
		);
	}
});

React.render(<Selectlist/>, document.body, () => {});