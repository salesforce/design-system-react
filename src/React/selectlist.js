// SELECTLIST CONTROL

// core
import {Landmark} from '../Landmark';
import {SelectlistCore} from '../SelectlistCore';

// framework specific
import React from 'react';
import DropdownButton from 'react-bootstrap/lib/DropdownButton';
import MenuItem from 'react-bootstrap/lib/MenuItem';

export var Selectlist = React.createClass(Object.assign({}, SelectlistCore, {	
	propTypes: {
		disabled: React.PropTypes.bool,
		selection: React.PropTypes.number,
		collection: React.PropTypes.array
	},
	
	getInitialState () {
		return this.__getInitialState();
	},
		
	menuItems () {
		return this.props.collection.map((menuItem) => {
			return <MenuItem eventKey={menuItem.id} onSelect={this.handleMenuItemClicked}>
					{menuItem.name}
				</MenuItem>;
		});
	},
	
	componentWillReceiveProps(nextProps) {
		this.__initializeOptions(nextProps);
	},
	
	render () {
		var selection = this.getSelection();
		
		return (
			<DropdownButton className={this.cssClasses.CONTROL} disabled={this.state.disabled} title={selection ? selection.name : 'None selected'} key={this.props.id}>{this.menuItems()}</DropdownButton>
		);
	},
	
	componentWillMount () {
		this.elements = {
			wrapper: {
				toggleClass: function (cssClass, state) {
					Landmark.log(cssClass, state);
				}
			}
		}
		
		this.__constructor(this.props);
	},
	
	handleMenuItemClicked (eventKey, href, target) {
		this.setSelectionByKey('id', eventKey);
	}
}));