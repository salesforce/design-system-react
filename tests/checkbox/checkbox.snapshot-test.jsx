import React from 'react';
import renderer from 'react-test-renderer';
import { renderMarkup } from '../snapshot-helpers';

import CheckboxBase from '../../examples/forms/checkbox/snapshot-base';
import CheckboxToggle from '../../examples/forms/checkbox/snapshot-toggle';


test('Checkbox Base DOM Snapshot', () => {
	const domTree = renderer.create(
		<CheckboxBase />,
	).toJSON();
	expect(domTree).toMatchSnapshot();
});


test('Checkbox Base HTML Snapshot', () => {
	expect(renderMarkup(CheckboxBase)).toMatchSnapshot();
});

test('Checkbox Toggle DOM Snapshot', () => {
	const domTree = renderer.create(
		<CheckboxToggle />,
	).toJSON();
	expect(domTree).toMatchSnapshot();
});


test('Checkbox Toggle HTML Snapshot', () => {
	expect(renderMarkup(CheckboxToggle)).toMatchSnapshot();
});

