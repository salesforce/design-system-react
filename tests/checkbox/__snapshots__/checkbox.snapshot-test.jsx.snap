// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox Base DOM Snapshot 1`] = `
<div
  className="slds-grid slds-grid--pull-padded slds-grid--vertical-align-center"
>
  <div
    className="slds-col--padded"
  >
    <div
      className="slds-form-element"
    >
      <div
        className="slds-form-element__control"
      >
        <span
          className="slds-checkbox"
        >
          <input
            aria-controls={undefined}
            aria-describedby={undefined}
            aria-owns={undefined}
            aria-required={undefined}
            checked={undefined}
            disabled={undefined}
            id="checkbox-base-example"
            name={undefined}
            onBlur={undefined}
            onChange={[Function]}
            onFocus={undefined}
            onKeyDown={undefined}
            onKeyPress={undefined}
            onKeyUp={undefined}
            required={undefined}
            role={undefined}
            type="checkbox"
          />
          <label
            className="slds-checkbox__label"
            htmlFor="checkbox-base-example"
          >
            <span
              className="slds-checkbox--faux"
            />
            <span
              className="slds-form-element__label"
            >
              Checkbox
            </span>
          </label>
        </span>
      </div>
    </div>
  </div>
  <div
    className="slds-col--padded"
  >
    <div
      className="slds-form-element slds-has-error"
    >
      <div
        className="slds-form-element__control"
      >
        <span
          className="slds-checkbox"
        >
          <input
            aria-controls={undefined}
            aria-describedby={undefined}
            aria-owns={undefined}
            aria-required={undefined}
            checked={undefined}
            disabled={undefined}
            id="checkbox-base-example--error"
            name={undefined}
            onBlur={undefined}
            onChange={[Function]}
            onFocus={undefined}
            onKeyDown={undefined}
            onKeyPress={undefined}
            onKeyUp={undefined}
            required={undefined}
            role={undefined}
            type="checkbox"
          />
          <label
            className="slds-checkbox__label"
            htmlFor="checkbox-base-example--error"
          >
            <span
              className="slds-checkbox--faux"
            />
            <span
              className="slds-form-element__label"
            >
              Checkbox
            </span>
          </label>
        </span>
      </div>
      <div
        className="slds-form-element__help"
      >
        This field has an error
      </div>
    </div>
  </div>
  <div
    className="slds-col--padded"
  >
    <div
      className="slds-form-element"
    >
      <div
        className="slds-form-element__control"
      >
        <span
          className="slds-checkbox"
        >
          <input
            aria-controls={undefined}
            aria-describedby={undefined}
            aria-owns={undefined}
            aria-required={undefined}
            checked={undefined}
            disabled={true}
            id="checkbox-base-example--disabled"
            name={undefined}
            onBlur={undefined}
            onChange={[Function]}
            onFocus={undefined}
            onKeyDown={undefined}
            onKeyPress={undefined}
            onKeyUp={undefined}
            required={undefined}
            role={undefined}
            type="checkbox"
          />
          <label
            className="slds-checkbox__label"
            htmlFor="checkbox-base-example--disabled"
          >
            <span
              className="slds-checkbox--faux"
            />
            <span
              className="slds-form-element__label"
            >
              Checkbox (disabled)
            </span>
          </label>
        </span>
      </div>
    </div>
  </div>
  <div
    className="slds-col--padded"
  >
    <div
      className="slds-form-element is-required"
    >
      <div
        className="slds-form-element__control"
      >
        <span
          className="slds-checkbox"
        >
          <abbr
            className="slds-required"
            title="required"
          >
            *
          </abbr>
          <input
            aria-controls={undefined}
            aria-describedby={undefined}
            aria-owns={undefined}
            aria-required={undefined}
            checked={undefined}
            disabled={undefined}
            id="checkbox-base-example--required"
            name={undefined}
            onBlur={undefined}
            onChange={[Function]}
            onFocus={undefined}
            onKeyDown={undefined}
            onKeyPress={undefined}
            onKeyUp={undefined}
            required={true}
            role={undefined}
            type="checkbox"
          />
          <label
            className="slds-checkbox__label"
            htmlFor="checkbox-base-example--required"
          >
            <span
              className="slds-checkbox--faux"
            />
            <span
              className="slds-form-element__label"
            >
              Checkbox (required)
            </span>
          </label>
        </span>
      </div>
    </div>
  </div>
  <div
    className="slds-col--padded"
  >
    <div
      className="slds-form-element"
    >
      <div
        className="slds-form-element__control"
      >
        <span
          className="slds-checkbox"
        >
          <input
            aria-controls={undefined}
            aria-describedby={undefined}
            aria-owns={undefined}
            aria-required={undefined}
            checked={undefined}
            disabled={undefined}
            id="checkbox-base-example--assistive-text"
            name={undefined}
            onBlur={undefined}
            onChange={[Function]}
            onFocus={undefined}
            onKeyDown={undefined}
            onKeyPress={undefined}
            onKeyUp={undefined}
            required={undefined}
            role={undefined}
            type="checkbox"
          />
          <label
            className="slds-checkbox__label"
            htmlFor="checkbox-base-example--assistive-text"
          >
            <span
              className="slds-checkbox--faux"
            />
            <span
              className="slds-assistive-text"
            >
              Checkbox (with assistive text)
            </span>
          </label>
        </span>
      </div>
    </div>
  </div>
  <div
    className="slds-col--padded"
  >
    <div
      className="slds-form-element"
    >
      <div
        className="slds-form-element__control"
      >
        <span
          className="slds-checkbox"
        >
          <input
            aria-controls={undefined}
            aria-describedby={undefined}
            aria-owns={undefined}
            aria-required={undefined}
            checked={true}
            disabled={undefined}
            id="checkbox-base-example--checked"
            name={undefined}
            onBlur={undefined}
            onChange={[Function]}
            onFocus={undefined}
            onKeyDown={undefined}
            onKeyPress={undefined}
            onKeyUp={undefined}
            required={undefined}
            role={undefined}
            type="checkbox"
          />
          <label
            className="slds-checkbox__label"
            htmlFor="checkbox-base-example--checked"
          >
            <span
              className="slds-checkbox--faux"
            />
            <span
              className="slds-form-element__label"
            >
              Checkbox (checked)
            </span>
          </label>
        </span>
      </div>
    </div>
  </div>
  <div
    className="slds-col--padded"
  >
    <div
      className="slds-form-element"
    >
      <div
        className="slds-form-element__control"
      >
        <span
          className="slds-checkbox"
        >
          <input
            aria-controls={undefined}
            aria-describedby={undefined}
            aria-owns={undefined}
            aria-required={undefined}
            checked={true}
            disabled={true}
            id="checkbox-base-example--checked-disabled"
            name={undefined}
            onBlur={undefined}
            onChange={[Function]}
            onFocus={undefined}
            onKeyDown={undefined}
            onKeyPress={undefined}
            onKeyUp={undefined}
            required={undefined}
            role={undefined}
            type="checkbox"
          />
          <label
            className="slds-checkbox__label"
            htmlFor="checkbox-base-example--checked-disabled"
          >
            <span
              className="slds-checkbox--faux"
            />
            <span
              className="slds-form-element__label"
            >
              Checkbox (checked + disabled)
            </span>
          </label>
        </span>
      </div>
    </div>
  </div>
  <div
    className="slds-col--padded"
  >
    <div
      className="slds-form-element"
    >
      <div
        className="slds-form-element__control"
      >
        <span
          className="slds-checkbox"
        >
          <input
            aria-controls={undefined}
            aria-describedby={undefined}
            aria-owns={undefined}
            aria-required={undefined}
            checked={undefined}
            disabled={undefined}
            id="checkbox-base-example--indeterminate"
            name={undefined}
            onBlur={undefined}
            onChange={[Function]}
            onFocus={undefined}
            onKeyDown={undefined}
            onKeyPress={undefined}
            onKeyUp={undefined}
            required={undefined}
            role={undefined}
            type="checkbox"
          />
          <label
            className="slds-checkbox__label"
            htmlFor="checkbox-base-example--indeterminate"
          >
            <span
              className="slds-checkbox--faux"
            />
            <span
              className="slds-form-element__label"
            >
              Checkbox (indeterminate)
            </span>
          </label>
        </span>
      </div>
    </div>
  </div>
</div>
`;

exports[`Checkbox Base HTML Snapshot 1`] = `
"<div class=\\"slds-grid slds-grid--pull-padded slds-grid--vertical-align-center\\">
  <div class=\\"slds-col--padded\\">
    <div class=\\"slds-form-element\\">
      <div class=\\"slds-form-element__control\\"><span class=\\"slds-checkbox\\"><input type=\\"checkbox\\" id=\\"checkbox-base-example\\"/><label class=\\"slds-checkbox__label\\" for=\\"checkbox-base-example\\"><span class=\\"slds-checkbox--faux\\"></span><span class=\\"slds-form-element__label\\">Checkbox</span></label>
        </span>
      </div>
    </div>
  </div>
  <div class=\\"slds-col--padded\\">
    <div class=\\"slds-form-element slds-has-error\\">
      <div class=\\"slds-form-element__control\\"><span class=\\"slds-checkbox\\"><input type=\\"checkbox\\" id=\\"checkbox-base-example--error\\"/><label class=\\"slds-checkbox__label\\" for=\\"checkbox-base-example--error\\"><span class=\\"slds-checkbox--faux\\"></span><span class=\\"slds-form-element__label\\">Checkbox</span></label>
        </span>
      </div>
      <div class=\\"slds-form-element__help\\">This field has an error</div>
    </div>
  </div>
  <div class=\\"slds-col--padded\\">
    <div class=\\"slds-form-element\\">
      <div class=\\"slds-form-element__control\\"><span class=\\"slds-checkbox\\"><input type=\\"checkbox\\" disabled=\\"\\" id=\\"checkbox-base-example--disabled\\"/><label class=\\"slds-checkbox__label\\" for=\\"checkbox-base-example--disabled\\"><span class=\\"slds-checkbox--faux\\"></span><span class=\\"slds-form-element__label\\">Checkbox (disabled)</span></label>
        </span>
      </div>
    </div>
  </div>
  <div class=\\"slds-col--padded\\">
    <div class=\\"slds-form-element is-required\\">
      <div class=\\"slds-form-element__control\\"><span class=\\"slds-checkbox\\"><abbr class=\\"slds-required\\" title=\\"required\\">*</abbr><input type=\\"checkbox\\" id=\\"checkbox-base-example--required\\" required=\\"\\"/><label class=\\"slds-checkbox__label\\" for=\\"checkbox-base-example--required\\"><span class=\\"slds-checkbox--faux\\"></span>
        <span
          class=\\"slds-form-element__label\\">Checkbox (required)</span>
          </label>
          </span>
      </div>
    </div>
  </div>
  <div class=\\"slds-col--padded\\">
    <div class=\\"slds-form-element\\">
      <div class=\\"slds-form-element__control\\"><span class=\\"slds-checkbox\\"><input type=\\"checkbox\\" id=\\"checkbox-base-example--assistive-text\\"/><label class=\\"slds-checkbox__label\\" for=\\"checkbox-base-example--assistive-text\\"><span class=\\"slds-checkbox--faux\\"></span><span class=\\"slds-assistive-text\\">Checkbox (with assistive text)</span></label>
        </span>
      </div>
    </div>
  </div>
  <div class=\\"slds-col--padded\\">
    <div class=\\"slds-form-element\\">
      <div class=\\"slds-form-element__control\\"><span class=\\"slds-checkbox\\"><input type=\\"checkbox\\" checked=\\"\\" id=\\"checkbox-base-example--checked\\"/><label class=\\"slds-checkbox__label\\" for=\\"checkbox-base-example--checked\\"><span class=\\"slds-checkbox--faux\\"></span><span class=\\"slds-form-element__label\\">Checkbox (checked)</span></label>
        </span>
      </div>
    </div>
  </div>
  <div class=\\"slds-col--padded\\">
    <div class=\\"slds-form-element\\">
      <div class=\\"slds-form-element__control\\"><span class=\\"slds-checkbox\\"><input type=\\"checkbox\\" disabled=\\"\\" checked=\\"\\" id=\\"checkbox-base-example--checked-disabled\\"/><label class=\\"slds-checkbox__label\\" for=\\"checkbox-base-example--checked-disabled\\"><span class=\\"slds-checkbox--faux\\"></span>
        <span
          class=\\"slds-form-element__label\\">Checkbox (checked + disabled)</span>
          </label>
          </span>
      </div>
    </div>
  </div>
  <div class=\\"slds-col--padded\\">
    <div class=\\"slds-form-element\\">
      <div class=\\"slds-form-element__control\\"><span class=\\"slds-checkbox\\"><input type=\\"checkbox\\" id=\\"checkbox-base-example--indeterminate\\"/><label class=\\"slds-checkbox__label\\" for=\\"checkbox-base-example--indeterminate\\"><span class=\\"slds-checkbox--faux\\"></span><span class=\\"slds-form-element__label\\">Checkbox (indeterminate)</span></label>
        </span>
      </div>
    </div>
  </div>
</div>"
`;

exports[`Checkbox Toggle DOM Snapshot 1`] = `
<div
  className="slds-grid slds-grid--pull-padded slds-grid--vertical-align-center"
>
  <div
    className="slds-col--padded"
  >
    <div
      className="slds-form-element"
    >
      <label
        className="slds-checkbox--toggle slds-grid"
        htmlFor="checkbox-toggle-example"
      >
        <span
          className="slds-form-element__label slds-m-bottom--none"
        >
          Toggle
        </span>
        <input
          aria-controls={undefined}
          aria-describedby="checkbox-toggle-example-desc"
          aria-owns={undefined}
          aria-required={undefined}
          checked={undefined}
          disabled={undefined}
          id="checkbox-toggle-example"
          name={undefined}
          onBlur={undefined}
          onChange={[Function]}
          onFocus={undefined}
          onKeyDown={undefined}
          onKeyPress={undefined}
          onKeyUp={undefined}
          required={undefined}
          role={undefined}
          type="checkbox"
        />
        <span
          aria-live="assertive"
          className="slds-checkbox--faux_container"
          id="checkbox-toggle-example-desc"
        >
          <span
            className="slds-checkbox--faux"
          />
          <span
            className="slds-checkbox--on"
          >
            Enabled
          </span>
          <span
            className="slds-checkbox--off"
          >
            Disabled
          </span>
        </span>
      </label>
    </div>
  </div>
  <div
    className="slds-col--padded"
  >
    <div
      className="slds-form-element slds-has-error"
    >
      <label
        className="slds-checkbox--toggle slds-grid"
        htmlFor="checkbox-toggle-example--error"
      >
        <span
          className="slds-form-element__label slds-m-bottom--none"
        >
          Toggle
        </span>
        <input
          aria-controls={undefined}
          aria-describedby="checkbox-toggle-example--error-desc"
          aria-owns={undefined}
          aria-required={undefined}
          checked={undefined}
          disabled={undefined}
          id="checkbox-toggle-example--error"
          name={undefined}
          onBlur={undefined}
          onChange={[Function]}
          onFocus={undefined}
          onKeyDown={undefined}
          onKeyPress={undefined}
          onKeyUp={undefined}
          required={undefined}
          role={undefined}
          type="checkbox"
        />
        <span
          aria-live="assertive"
          className="slds-checkbox--faux_container"
          id="checkbox-toggle-example--error-desc"
        >
          <span
            className="slds-checkbox--faux"
          />
          <span
            className="slds-checkbox--on"
          >
            Enabled
          </span>
          <span
            className="slds-checkbox--off"
          >
            Disabled
          </span>
        </span>
      </label>
      <div
        className="slds-form-element__help"
      >
        This field has an error
      </div>
    </div>
  </div>
  <div
    className="slds-col--padded"
  >
    <div
      className="slds-form-element"
    >
      <label
        className="slds-checkbox--toggle slds-grid"
        htmlFor="checkbox-toggle-example--disabled"
      >
        <span
          className="slds-form-element__label slds-m-bottom--none"
        >
          Toggle (disabled)
        </span>
        <input
          aria-controls={undefined}
          aria-describedby="checkbox-toggle-example--disabled-desc"
          aria-owns={undefined}
          aria-required={undefined}
          checked={undefined}
          disabled={true}
          id="checkbox-toggle-example--disabled"
          name={undefined}
          onBlur={undefined}
          onChange={[Function]}
          onFocus={undefined}
          onKeyDown={undefined}
          onKeyPress={undefined}
          onKeyUp={undefined}
          required={undefined}
          role={undefined}
          type="checkbox"
        />
        <span
          aria-live="assertive"
          className="slds-checkbox--faux_container"
          id="checkbox-toggle-example--disabled-desc"
        >
          <span
            className="slds-checkbox--faux"
          />
          <span
            className="slds-checkbox--on"
          >
            Enabled
          </span>
          <span
            className="slds-checkbox--off"
          >
            Disabled
          </span>
        </span>
      </label>
    </div>
  </div>
  <div
    className="slds-col--padded"
  >
    <div
      className="slds-form-element is-required"
    >
      <label
        className="slds-checkbox--toggle slds-grid"
        htmlFor="checkbox-toggle-example--required"
      >
        <abbr
          className="slds-required"
          title="required"
        >
          *
        </abbr>
        <span
          className="slds-form-element__label slds-m-bottom--none"
        >
          Toggle (required)
        </span>
        <input
          aria-controls={undefined}
          aria-describedby="checkbox-toggle-example--required-desc"
          aria-owns={undefined}
          aria-required={undefined}
          checked={undefined}
          disabled={undefined}
          id="checkbox-toggle-example--required"
          name={undefined}
          onBlur={undefined}
          onChange={[Function]}
          onFocus={undefined}
          onKeyDown={undefined}
          onKeyPress={undefined}
          onKeyUp={undefined}
          required={true}
          role={undefined}
          type="checkbox"
        />
        <span
          aria-live="assertive"
          className="slds-checkbox--faux_container"
          id="checkbox-toggle-example--required-desc"
        >
          <span
            className="slds-checkbox--faux"
          />
          <span
            className="slds-checkbox--on"
          >
            Enabled
          </span>
          <span
            className="slds-checkbox--off"
          >
            Disabled
          </span>
        </span>
      </label>
    </div>
  </div>
  <div
    className="slds-col--padded"
  >
    <div
      className="slds-form-element"
    >
      <label
        className="slds-checkbox--toggle slds-grid"
        htmlFor="checkbox-toggle-example--assitive-text"
      >
        <span
          className="slds-assistive-text"
        >
          Toggle (with assistive text)
        </span>
        <input
          aria-controls={undefined}
          aria-describedby="checkbox-toggle-example--assitive-text-desc"
          aria-owns={undefined}
          aria-required={undefined}
          checked={undefined}
          disabled={undefined}
          id="checkbox-toggle-example--assitive-text"
          name={undefined}
          onBlur={undefined}
          onChange={[Function]}
          onFocus={undefined}
          onKeyDown={undefined}
          onKeyPress={undefined}
          onKeyUp={undefined}
          required={undefined}
          role={undefined}
          type="checkbox"
        />
        <span
          aria-live="assertive"
          className="slds-checkbox--faux_container"
          id="checkbox-toggle-example--assitive-text-desc"
        >
          <span
            className="slds-checkbox--faux"
          />
          <span
            className="slds-checkbox--on"
          >
            Enabled
          </span>
          <span
            className="slds-checkbox--off"
          >
            Disabled
          </span>
        </span>
      </label>
    </div>
  </div>
  <div
    className="slds-col--padded"
  >
    <div
      className="slds-form-element"
    >
      <label
        className="slds-checkbox--toggle slds-grid"
        htmlFor="checkbox-toggle-example--checked"
      >
        <span
          className="slds-form-element__label slds-m-bottom--none"
        >
          Toggle (checked)
        </span>
        <input
          aria-controls={undefined}
          aria-describedby="checkbox-toggle-example--checked-desc"
          aria-owns={undefined}
          aria-required={undefined}
          checked={true}
          disabled={undefined}
          id="checkbox-toggle-example--checked"
          name={undefined}
          onBlur={undefined}
          onChange={[Function]}
          onFocus={undefined}
          onKeyDown={undefined}
          onKeyPress={undefined}
          onKeyUp={undefined}
          required={undefined}
          role={undefined}
          type="checkbox"
        />
        <span
          aria-live="assertive"
          className="slds-checkbox--faux_container"
          id="checkbox-toggle-example--checked-desc"
        >
          <span
            className="slds-checkbox--faux"
          />
          <span
            className="slds-checkbox--on"
          >
            Enabled
          </span>
          <span
            className="slds-checkbox--off"
          >
            Disabled
          </span>
        </span>
      </label>
    </div>
  </div>
  <div
    className="slds-col--padded"
  >
    <div
      className="slds-form-element"
    >
      <label
        className="slds-checkbox--toggle slds-grid"
        htmlFor="checkbox-toggle-example--checked-disabled"
      >
        <span
          className="slds-form-element__label slds-m-bottom--none"
        >
          Toggle (checked + disabled)
        </span>
        <input
          aria-controls={undefined}
          aria-describedby="checkbox-toggle-example--checked-disabled-desc"
          aria-owns={undefined}
          aria-required={undefined}
          checked={true}
          disabled={true}
          id="checkbox-toggle-example--checked-disabled"
          name={undefined}
          onBlur={undefined}
          onChange={[Function]}
          onFocus={undefined}
          onKeyDown={undefined}
          onKeyPress={undefined}
          onKeyUp={undefined}
          required={undefined}
          role={undefined}
          type="checkbox"
        />
        <span
          aria-live="assertive"
          className="slds-checkbox--faux_container"
          id="checkbox-toggle-example--checked-disabled-desc"
        >
          <span
            className="slds-checkbox--faux"
          />
          <span
            className="slds-checkbox--on"
          >
            Enabled
          </span>
          <span
            className="slds-checkbox--off"
          >
            Disabled
          </span>
        </span>
      </label>
    </div>
  </div>
</div>
`;

exports[`Checkbox Toggle HTML Snapshot 1`] = `
"<div class=\\"slds-grid slds-grid--pull-padded slds-grid--vertical-align-center\\">
  <div class=\\"slds-col--padded\\">
    <div class=\\"slds-form-element\\"><label class=\\"slds-checkbox--toggle slds-grid\\" for=\\"checkbox-toggle-example\\"><span class=\\"slds-form-element__label slds-m-bottom--none\\">Toggle</span><input type=\\"checkbox\\" aria-describedby=\\"checkbox-toggle-example-desc\\" id=\\"checkbox-toggle-example\\"/><span id=\\"checkbox-toggle-example-desc\\" class=\\"slds-checkbox--faux_container\\" aria-live=\\"assertive\\"><span class=\\"slds-checkbox--faux\\"></span><span class=\\"slds-checkbox--on\\">Enabled</span><span class=\\"slds-checkbox--off\\">Disabled</span></span></label></div>
  </div>
  <div class=\\"slds-col--padded\\">
    <div class=\\"slds-form-element slds-has-error\\"><label class=\\"slds-checkbox--toggle slds-grid\\" for=\\"checkbox-toggle-example--error\\"><span class=\\"slds-form-element__label slds-m-bottom--none\\">Toggle</span><input type=\\"checkbox\\" aria-describedby=\\"checkbox-toggle-example--error-desc\\" id=\\"checkbox-toggle-example--error\\"/><span id=\\"checkbox-toggle-example--error-desc\\" class=\\"slds-checkbox--faux_container\\" aria-live=\\"assertive\\"><span class=\\"slds-checkbox--faux\\"></span><span class=\\"slds-checkbox--on\\">Enabled</span><span class=\\"slds-checkbox--off\\">Disabled</span></span></label>
      <div
        class=\\"slds-form-element__help\\">This field has an error</div>
  </div>
</div>
<div class=\\"slds-col--padded\\">
  <div class=\\"slds-form-element\\"><label class=\\"slds-checkbox--toggle slds-grid\\" for=\\"checkbox-toggle-example--disabled\\"><span class=\\"slds-form-element__label slds-m-bottom--none\\">Toggle (disabled)</span><input type=\\"checkbox\\" aria-describedby=\\"checkbox-toggle-example--disabled-desc\\" disabled=\\"\\" id=\\"checkbox-toggle-example--disabled\\"/><span id=\\"checkbox-toggle-example--disabled-desc\\" class=\\"slds-checkbox--faux_container\\" aria-live=\\"assertive\\"><span class=\\"slds-checkbox--faux\\"></span><span class=\\"slds-checkbox--on\\">Enabled</span><span class=\\"slds-checkbox--off\\">Disabled</span></span></label></div>
</div>
<div class=\\"slds-col--padded\\">
  <div class=\\"slds-form-element is-required\\"><label class=\\"slds-checkbox--toggle slds-grid\\" for=\\"checkbox-toggle-example--required\\"><abbr class=\\"slds-required\\" title=\\"required\\">*</abbr><span class=\\"slds-form-element__label slds-m-bottom--none\\">Toggle (required)</span><input type=\\"checkbox\\" aria-describedby=\\"checkbox-toggle-example--required-desc\\" id=\\"checkbox-toggle-example--required\\" required=\\"\\"/><span id=\\"checkbox-toggle-example--required-desc\\" class=\\"slds-checkbox--faux_container\\" aria-live=\\"assertive\\"><span class=\\"slds-checkbox--faux\\"></span><span class=\\"slds-checkbox--on\\">Enabled</span><span class=\\"slds-checkbox--off\\">Disabled</span></span></label></div>
</div>
<div class=\\"slds-col--padded\\">
  <div class=\\"slds-form-element\\"><label class=\\"slds-checkbox--toggle slds-grid\\" for=\\"checkbox-toggle-example--assitive-text\\"><span class=\\"slds-assistive-text\\">Toggle (with assistive text)</span><input type=\\"checkbox\\" aria-describedby=\\"checkbox-toggle-example--assitive-text-desc\\" id=\\"checkbox-toggle-example--assitive-text\\"/><span id=\\"checkbox-toggle-example--assitive-text-desc\\" class=\\"slds-checkbox--faux_container\\" aria-live=\\"assertive\\"><span class=\\"slds-checkbox--faux\\"></span><span class=\\"slds-checkbox--on\\">Enabled</span><span class=\\"slds-checkbox--off\\">Disabled</span></span></label></div>
</div>
<div class=\\"slds-col--padded\\">
  <div class=\\"slds-form-element\\"><label class=\\"slds-checkbox--toggle slds-grid\\" for=\\"checkbox-toggle-example--checked\\"><span class=\\"slds-form-element__label slds-m-bottom--none\\">Toggle (checked)</span><input type=\\"checkbox\\" aria-describedby=\\"checkbox-toggle-example--checked-desc\\" id=\\"checkbox-toggle-example--checked\\" checked=\\"\\"/><span id=\\"checkbox-toggle-example--checked-desc\\" class=\\"slds-checkbox--faux_container\\" aria-live=\\"assertive\\"><span class=\\"slds-checkbox--faux\\"></span><span class=\\"slds-checkbox--on\\">Enabled</span><span class=\\"slds-checkbox--off\\">Disabled</span></span></label></div>
</div>
<div class=\\"slds-col--padded\\">
  <div class=\\"slds-form-element\\"><label class=\\"slds-checkbox--toggle slds-grid\\" for=\\"checkbox-toggle-example--checked-disabled\\"><span class=\\"slds-form-element__label slds-m-bottom--none\\">Toggle (checked + disabled)</span><input type=\\"checkbox\\" aria-describedby=\\"checkbox-toggle-example--checked-disabled-desc\\" disabled=\\"\\" id=\\"checkbox-toggle-example--checked-disabled\\" checked=\\"\\"/><span id=\\"checkbox-toggle-example--checked-disabled-desc\\" class=\\"slds-checkbox--faux_container\\" aria-live=\\"assertive\\"><span class=\\"slds-checkbox--faux\\"></span><span class=\\"slds-checkbox--on\\">Enabled</span><span class=\\"slds-checkbox--off\\">Disabled</span></span></label></div>
</div>
</div>"
`;
